src/components/networked.js

```diff
@@ -388,19 +388,18 @@ AFRAME.registerComponent('networked', {
   },
 
   updateNetworkedComponents: function(components) {
-    for (var componentIndex in components) {
+    for (var componentIndex = 0, l = this.componentSchemas.length; componentIndex < l; componentIndex++) {
       var componentData = components[componentIndex];
       var componentSchema = this.componentSchemas[componentIndex];
       var componentElement = this.getCachedElement(componentIndex);
 
-      if (componentElement === null || componentData === null ) {
+      if (componentElement === null || componentData === undefined ) {
```

Comment:
Since you changed this to loop over `componentSchemas`, do we now need to guard against `componentData === null` or `componentData === undefined` (depending on whether `gatherComponentData` was passed `true` or `false` respectively?) 
I don't have a firm grasp of the logic here but thought that `null` check was important.

Reply:
(Also note that we do not just want to check `!componentData` because `false` is a valid value for it to have!)

Reply:
ah good call -- semantically it looks like it was an expected condition that the key could exist but be `null`